cpp/tests/test_save_results.cpp

```diff
@@ -105,3 +113,188 @@ TEST(TestSaveResult, compareResultWithH5)
+        model.populations[{i, mio::osecir::InfectionState::InfectedNoSymptoms}] = fact * num_car_t0;
+        model.populations[{i, mio::osecir::InfectionState::InfectedSymptoms}]   = fact * num_inf_t0;
+        model.populations[{i, mio::osecir::InfectionState::InfectedSevere}]     = fact * num_hosp_t0;
+        model.populations[{i, mio::osecir::InfectionState::InfectedCritical}]   = fact * num_icu_t0;
+        model.populations[{i, mio::osecir::InfectionState::Recovered}]          = fact * num_rec_t0;
+        model.populations[{i, mio::osecir::InfectionState::Dead}]               = fact * num_dead_t0;
+        model.populations.set_difference_from_group_total<mio::AgeGroup>({i, mio::osecir::InfectionState::Susceptible},
+                                                                         fact * num_total_t0);
+
+        params.get<mio::osecir::TransmissionProbabilityOnContact>()[i] = 0.05;
+        params.get<mio::osecir::RelativeTransmissionNoSymptoms>()[i]   = 0.67;
+        params.get<mio::osecir::RecoveredPerInfectedNoSymptoms>()[i]   = 0.09;
+        params.get<mio::osecir::RiskOfInfectionFromSymptomatic>()[i]   = 0.25;
+        params.get<mio::osecir::SeverePerInfectedSymptoms>()[i]        = 0.2;
+        params.get<mio::osecir::CriticalPerSevere>()[i]                = 0.25;
+        params.get<mio::osecir::DeathsPerCritical>()[i]                = 0.3;
+    }
+
+    mio::ContactMatrixGroup& contact_matrix = params.get<mio::osecir::ContactPatterns>();
+    contact_matrix[0] = mio::ContactMatrix(Eigen::MatrixXd::Constant(num_groups, num_groups, fact * cont_freq));
+
+    mio::osecir::set_params_distributions_normal(model, t0, tmax, 0.2);
+
+    auto graph = mio::Graph<mio::osecir::Model, mio::MigrationParameters>();
+    graph.add_node(0, model);
+    graph.add_node(1, model);
+    graph.add_edge(0, 1, mio::MigrationParameters(Eigen::VectorXd::Constant(Eigen::Index(num_groups * 8), 1.0)));
+
+    auto num_runs = 3;
+    auto study    = mio::ParameterStudy<mio::osecir::Simulation<>>(graph, 0.0, 2.0, 0.5, num_runs);
+    auto results  = study.run([](auto&& g) {
+        return draw_sample(g);
+    });
+
+    TempFileRegister tmp_file_register;
+    std::string tmp_results_dir = tmp_file_register.get_unique_path();
+    ASSERT_THAT(mio::create_directory(tmp_results_dir), IsSuccess());
+
+    auto ensemble_results = std::vector<std::vector<mio::TimeSeries<double>>>{};
+    ensemble_results.reserve(size_t(num_runs));
+    auto ensemble_params = std::vector<std::vector<mio::osecir::Model>>{};
+    ensemble_params.reserve(size_t(num_runs));
+    auto save_result_status = mio::IOResult<void>(mio::success());
+    auto run_idx            = size_t(0);
+    study.run(
+        [](auto&& g) {
+            return draw_sample(g);
+        },
+        [&](auto results_graph) {
+            ensemble_results.push_back(mio::interpolate_simulation_result(results_graph));
+
+            ensemble_params.emplace_back();
```

Comment:
What do you need this empty emplace_back() for?

Reply:
Create an object at the end of the vector ensemble_params. Otherwise we would get a segmentation error.